services/motion/builtin/move_request.go

```diff
@@ -18,11 +18,18 @@ import (
 	"go.viam.com/rdk/spatialmath"
 )
 
+const (
+	defaultReplanCostFactor = 1.0
+	defaultMaxReplans       = -1 // Values below zero will replan infinitely
+)
+
 // moveRequest is a structure that contains all the information necessary for to make a move call.
 type moveRequest struct {
-	config        *motion.MotionConfiguration
-	planRequest   *motionplan.PlanRequest
-	kinematicBase kinematicbase.KinematicBase
+	config           *motion.MotionConfiguration
+	planRequest      *motionplan.PlanRequest
+	seedPlan         motionplan.Plan
```

Comment:
why do we need this seed plan?

Reply:
Technically we don't for this PR, but this PR will be immediately followed by [RSDK-3236](https://viam.atlassian.net/browse/RSDK-3236) which will need this.

Reply:
If we don't need it in this PR could we just add it in the second one?

Reply:
We can; happy to take this out if you prefer.

If I have a series of PRs that I know for certain will immediately follow one another, I find it helpful to wireframe out some of the functionality in later PRs in the earlier ones, to ensure that any architecture chosen in an earlier PR fits the future work nicely and prevents large immediate rewrites of the new code.

Reply:
I'm asking as requiring the caller of newMoveOnGlobeRequest to mutate record & set the move request's `seedPlan` before & after the call seems unintuitive to me & indicative that the api might not yet be fully fleshed out.

If it is not needed for this PR to work then I'm inclined to remove the code that is not needed in this ticket which might benefit from some api improvements.

https://github.com/viamrobotics/rdk/pull/3054/files/794af6a03f4f7adc21f648ca0f40ba71d6b32896#diff-01573c2337c9ee8bafdab389eacebcbd7c16692bd71bec1208ead3665f40920fR350-R356

Reply:
Note that the highlighted code you linked was removed :)

Reply:
fair enough, never mind. I didn't see that it had been removed. This can stay.